### ISSUE TITLE
Theme migration progress

### ISSUE THREAD

**Issue by morpheusthewhite** [morpheusthewhite/spicetify-themes#368]
Themes need to be updated to support new UI and Spicetify and branch `v2` has been created in this repository for the purpose.

Here I will list all the themes and the update progress, tagging maintainers of the existing who are able and want to contribute. As some themes in this repository are very similar they can be merged into a single updated theme with multiple color schemes. Please refer to the README in the new branch for all the information

- [x] Default (from Adapta-Nokto, Ark-Dark, Ark-Green, Blueballs, DeepCoral, Discord, Dobbo, Flatten, Gruvbox-Gold, Kaapi, Matcha, Midnight-Light, NightOwl, Phosphoria, Vaporwave, YoutubeDark) 
- [x] Sleek (from <s>BIB-Green</s>, Black, BladeRunner, <s>Challenger-Deep</s>, Cherry, DanDrumStoneNew, Dark, Dracula, Elementary, Gradianto, JarvisBot, <s>Material-Ocean</s>, MoonChild, Night, NightMoon, Nord, NoSleep, OneDarkish, Otto, OutrunDark, OldSchoolSenn, Pop-Dark, Twasi, SolarizedDark, Sweet, Tidal, TrekyGoldenrod, TychoAwake, WintergatanBlueprint)
- [x] BurntSienna (from BurntSienna, @pjaspinski)
- [x] Dribbblish (from Dribbblish, DribbblishDynamic, @khanhas @JulienMaille)
- [x] Onepunch (from Onepunch, @okarin001)
- [ ] ShadowCustom (from ShadowCustom, @CallMeVorex)
- [ ] Spicy (from Spicy)
- [x] Turntable (from Turntable, @grasonchan)

Please mention the people who contributed to color schemes and themes in each theme README.

**Comment by Milo123459** [comment]
Will start working on them.

**Comment by RootBurner** [comment]
Can users help in updating the themes?
I've mapped the texts to their respective colors and can be used to update some themes.
Ive already updated some from Samurai and Nord themes.

Here is the keymapping
; text               = main text, playlist names in main field, name of playlist selected in sidebar, headings
; subtext            = text in main buttons in sidebar, playlist names in sidebar, artist names, and mini infos
; main               = main field or main bg
; sidebar            = sidebar bg
; player             = player bg
; card               = card bg
; shadow             = bg of buttons like account, pop-up lyrics, full app display in main field
; selected row       = color of the song selected, scrollbar, caption and details of playlist, download and options button
; button             = playlist buttons bg in sidebar, drop-down menus, now playing song, play button bg, like button
; button-active      = hover on song selected
; button-disabled    = seekbar bg, volume bar bg
; tab-active         = button bg in main field (playlists,podcasts,artists,albums)
; notification       = notification
; notification-error = error
; misc               = misc

**Comment by RootBurner** [comment]
Made a better sheet for current themers for easy upgrade.

¯\\__(ツ)_/¯ = Colors that need testing with other colors

[Sheet]
New
text                         = main_fg
subtext                   = secondary_fg
main                       = main_bg
sidebar                   = sidebar_and_player_bg
player                     = sidebar_and_player_bg
card                        = ¯\\_ _(ツ)_ /¯ color of card 
shadow                   = cover_overlay_and_shadow 
selected-row           = ¯\\_ _(ツ)_ /¯ color of the selected song 
button                     = ¯\\_ _(ツ)_ /¯ confusion when placing old colors                           
button-active          = pressing_button_fg                    
button-disabled      = slider_bg
tab-active                = selected_button
notification              = ¯\\_ _(ツ)_ /¯ haven't seen any change
notification-error     = ¯\\_ _(ツ)_ /¯  
misc                         = ¯\\_ _(ツ)__/¯  

Correct if wrong.
Some colors may not be correct but it works for some dual tone themes.

**Comment by Milo123459** [comment]
What about changing the CSS? Anyone got any guides for that?

**Comment by morpheusthewhite** [comment]
@Milo123459 there isn't for sure about v2 (as far as I know) but it would be very nice to have something like that as well in the repository

**Comment by Milo123459** [comment]
Alrighty. Thanks.

**Comment by morpheusthewhite** [comment]
> Can users help in updating the themes?

I guess it depends if the previous maintainers want to update their themes on their own. Let us wait some time to see if they get to reply. 

Of the above-listed themes, Spicy has no real maintainer, as I pushed in when creating the repository, taking it from Spectrum chat, so feel free to update it and open a PR.

**Comment by JulienMaille** [comment]
Sorry but porting DribblishDynamic is out of my league :(

**Comment by RootBurner** [comment]
> @Milo123459 there isn't for sure about v2 (as far as I know) but it would be very nice to have something like that as well in the repository

It would be a good idea if some one could list out css commands to edit the interface in an easy way in the repository.

**Comment by okarin001** [comment]
it's done. I have commented on my user.css, as much as I could. So, if anyone needs any reference for the new UI can check that.

**Comment by morpheusthewhite** [comment]
> Sorry but porting DribblishDynamic is out of my league :(

I opened #377 about that, if anyone wants to contribute

**Comment by canbeardig** [comment]
@RootBurner In the sceond screenshot, how did you get rid of [the new gradient](https://github.com/khanhas/spicetify-cli/issues/637)?

**Comment by RootBurner** [comment]
I haven't come across gradients. I only made edits in the color.ini file. The only edit I did in the css files were the play button appearing white as default.

**Comment by canbeardig** [comment]
It seems it happens after v2.0 upgrade. Thanks for the reply.

**Comment by Milo123459** [comment]
Bug: Aritim-Dark & Arc-Dark are broken
![img](https://salvage.is-inside.me/39uZ88Tr.png)

**Comment by Milo123459** [comment]
^ Happened again. It seems to be broken and can't apply themes.

**Comment by RootBurner** [comment]
Ark Dark Base
![Screenshot 2021-05-22 145541](https://user-images.githubusercontent.com/84371526/119221634-dc4dfe80-bb0d-11eb-8dd1-f89298760cec.png)

**Comment by Zactheepic15** [comment]
Does anyone plan on porting [black](https://github.com/morpheusthewhite/spicetify-themes/tree/master/Black)? If not I could probably try and figure out the colors by myself, but not quite sure how I would port the user.css

**Comment by morpheusthewhite** [comment]
> Does anyone plan on porting [black](https://github.com/morpheusthewhite/spicetify-themes/tree/master/Black)? If not I could probably try and figure out the colors by myself, but not quite sure how I would port the user.css

There a Pull request (#370 ) for merging a theme on which we could push the Black color scheme. It will soon be merged

**Comment by morpheusthewhite** [comment]
@khanhas do you plan to update Dribbblish to v2?

**Comment by khanhas** [comment]
@morpheusthewhite I do but I'm currently busy with uni right now. Maybe 2,3 weeks later

**Comment by BlueDevil** [comment]
Would love to have Flatten ported to v2. Or I guess I could look into it myself.
Sleek BIB works great so far though :D

**Comment by JulienMaille** [comment]
> @morpheusthewhite I do but I'm currently busy with uni right now. Maybe 2,3 weeks later

I like how @khanhas always under-promise and over-deliver!
I was able to "patch" dribbblish into dribbblish-dynamic with just on additional js file, still wondering how to distribute this. I think the "light/dark" switch could merde with the original dribbblish.

**Comment by JulienMaille** [comment]
@morpheusthewhite would you accept adding v2 themes as submodules (and update them when a clean release is ready) rather than as code itself?
A preview of dribbblish-dynamic is available here: https://github.com/JulienMaille/dribbblish-dynamic-theme

**Comment by morpheusthewhite** [comment]
> @morpheusthewhite would you accept adding v2 themes as submodules (and update them when a clean release is ready) rather than as code itself?

I'm not sure using a submodule is a good idea because probably many users, especially on Windows, just download the zip file of this repository

**Comment by morpheusthewhite** [comment]
You can push it once it is ready directly on Dribbblish since khanhas agreed

**Comment by JulienMaille** [comment]
@morpheusthewhite @khanhas i would enjoy some feedback on the latest commit with color transitions (a frequent request) does it lag on your system?

**Comment by ericdrgn** [comment]
@JulienMaille it works great for me, I used the previous version last night too and now the transitions are much better. Awesome work! Honestly from everyone making this migration happen quick, but this was the theme I used so thanks.

**Comment by JulienMaille** [comment]
I could use some feedback on the bash script, calling all linux users!
`curl -fsSL https://raw.githubusercontent.com/JulienMaille/dribbblish-dynamic-theme/master/install.sh | sh`

**Comment by stpnwf** [comment]
> > @morpheusthewhite I do but I'm currently busy with uni right now. Maybe 2,3 weeks later
> 
> I like how @khanhas always under-promise and over-deliver!
> I was able to "patch" dribbblish into dribbblish-dynamic with just on additional js file, still wondering how to distribute this. I think the "light/dark" switch could merge with the original dribbblish.
> ![Dark](https://user-images.githubusercontent.com/182520/120538139-69366900-c3e6-11eb-85ca-6a0c3e2039a9.gif)

Does Dribbblish V2 support the "light/dark" switch by default? or must one install the version from your repo to get that feature?

**Comment by JulienMaille** [comment]
Currently you need to install the version from my repos.

**Comment by stpnwf** [comment]
@JulienMaille 
I Installed the one from your repo (from the script), but i had an issue where I could not read some of the text because of the background color. I noticed that playing songs seemed to place the album artwork as the background of the window, however the colors would not change according to the song played. I got a screenshot, but, since I tested this on a Linux Live USB, I forgot to save it when I rebooted to test @morpheusthewhite's version. :s
I can create an issue in your repo if you would like.

I was able to test what I wanted though: the toggle. I would love to be able to switch between two static color_schemes on the non dynamic Dribbblish.

---
@morpheusthewhite 
I also tested the version from this repo on a live environment. And I was able to install the latest version of spicetify (v 2.2.5) and spotify, (1.1.55) but nothing happened to my Spotify. It had the default look on even though I had the Dribbblish theme on. I copied and pasted the install instructions from the README file, so IDK what the problem could have been.
Please let me know if you would like me to open an issue with you.

I have the latest Spotify version installed (1.1.**55**.498.gf9a83c60).  Mine is from Spotify's Debian repo, but I also checked that the [Flathub](https://flathub.org/apps/details/com.spotify.Client) version is the same. 

README from Dribbblish lists: "Make sure you are using spicetify v2 and Spotify **v1.1.58 or later**." Is that a typo or Linux is not yet supported bc 1.1.58 is not out for it yet?

**Comment by ps13g** [comment]
> I have the latest Spotify version installed (1.1.**55**.498.gf9a83c60). Mine is from Spotify's Debian repo, but I also checked that the [Flathub](https://flathub.org/apps/details/com.spotify.Client) version is the same.
> 
> README from Dribbblish lists: "Make sure you are using spicetify v2 and Spotify **v1.1.58 or later**." Is that a typo or Linux is not yet supported bc 1.1.58 is not out for it yet?

I actually had the same issue a few days ago on MacOS and basically what needs to happen is that uninstall both Spotify and Spicetify entirely and then reinstall the programs again. After you reinstall make sure to copy over everything from the Zip File and then retry the entire process again of copy and pasting the code and it should work.

**Comment by stpnwf** [comment]
> what needs to happen is that uninstall both Spotify and Spicetify entirely and then reinstall the programs again.

Did you have a clean install before? I booted into a live Ubuntu USB and fresh installed everything to test it. I will wait a week or so and try again

**Comment by ps13g** [comment]
> > what needs to happen is that uninstall both Spotify and Spicetify entirely and then reinstall the programs again.
> 
> Did you have a clean install before? I booted into a live Ubuntu USB and fresh installed everything to test it. I will wait a week or so and try again

Yes I had to clean install it onto the OS and it worked for me at the end

**Comment by Tuuuukka** [comment]
So is anyone working on the Nord theme at the moment?

**Comment by morpheusthewhite** [comment]
@Tuuuukka as of plans it will be just a colour scheme for the Sleek theme

**Comment by morpheusthewhite** [comment]
We will switch branch in a week, I think now most of the people are using the new Spotify UI

**Comment by morpheusthewhite** [comment]
Just updated the branches. 

`v2` is now named `master`

If you have a local clone, you can update it by running:

```
git branch -m v2 master
git fetch origin
git branch -u origin/master master
git remote set-head origin -a
```
